docs/foreign-entity-ids.wiki

```diff
@@ -0,0 +1,31 @@
+EntityId contains the logical name of the repository the ID refers to. This serves as a namespace mechanism for entity IDs.
+EntityIds from repositories other than local repository are called '''foreign''' EntityIds,
+and, similarly, repositories other than local are '''foreign''' repositories.
+
+Notes:
+* The name used to refer to a given repository is local, and can differ from client to client.
+* The empty repository name "" refers to the local wiki.
+* Repository names can be used to look up the repository Site object, which in turn allows the ID to be mapped to the currect URI, URL, or API path for the repository it belongs to.
+* The serialization of the EntityId follows the pattern <code><repository>:<id></code>. If <code><repository></code> is empty, the leading <code>:</code> is optional. This format follows the convention set by XML namespaces, RDF prefixes, and MediaWiki interwiki links.
+* Repository names can be mapped during serialization and deserialization.
+* When reading data from a another repository, repository names in entity IDs get mapped from the names used on the other repository to the names used on the local wiki. In particular, the empty repository prefix is mapped to the name of the foreign repository while reading IDs.
+* When writing data to a another repository, repository names in entity IDs get mapped from the names used on the local wiki to the names used on the foreign repository. In particular, the foreign wiki's name is mapped to the empty prefix.
```

Comment:
I guess you're right, this part is not really relevant for what we're planning to do (at least in first iteration). I believe the description still makes sense but I will remove it for know, so it does not create confusion ("Why you throw exception here when docs define how to write stuffs to foreign repo?")